converter/convert.go

```diff
@@ -188,92 +204,57 @@ func makeSortedImages(images map[string]bundle.Image) []string {
 	return result
 }
 
-func parseManifests(descriptors []ocischemav1.Descriptor, into *bundle.Bundle, originRepo reference.Named) error {
-	for _, d := range descriptors {
-		var imageType string
-		switch d.MediaType {
-		case ocischemav1.MediaTypeImageManifest, ocischemav1.MediaTypeImageIndex:
-			imageType = "oci"
-		case images.MediaTypeDockerSchema2Manifest, images.MediaTypeDockerSchema2ManifestList:
-			imageType = "docker"
-		default:
-			return fmt.Errorf("unsupported manifest descriptor %q with mediatype %q", d.Digest, d.MediaType)
-		}
-		descriptorType, ok := d.Annotations[CNABDescriptorTypeAnnotation]
-		if !ok {
-			return fmt.Errorf("manifest descriptor %q has no CNAB descriptor type annotation %q", d.Digest, CNABDescriptorTypeAnnotation)
-		}
-		if descriptorType == CNABDescriptorTypeConfig {
-			continue
-		}
-		// strip tag/digest from originRepo
-		originRepo, err := reference.ParseNormalizedNamed(originRepo.Name())
-		if err != nil {
-			return fmt.Errorf("failed to create a digested reference for manifest descriptor %q: %s", d.Digest, err)
-		}
-		ref, err := reference.WithDigest(originRepo, d.Digest)
-		if err != nil {
-			return fmt.Errorf("failed to create a digested reference for manifest descriptor %q: %s", d.Digest, err)
-		}
-		refFamiliar := reference.FamiliarString(ref)
-		switch descriptorType {
-		// The current descriptor is an invocation image
-		case CNABDescriptorTypeInvocation:
-			into.InvocationImages = append(into.InvocationImages, bundle.InvocationImage{
-				BaseImage: bundle.BaseImage{
-					Image:     refFamiliar,
-					ImageType: imageType,
-					MediaType: d.MediaType,
-					Size:      uint64(d.Size),
-				},
-			})
-		// The current descriptor is a component image
-		case CNABDescriptorTypeComponent:
-			componentName, ok := d.Annotations[CNABDescriptorComponentNameAnnotation]
-			if !ok {
-				return fmt.Errorf("component name missing in descriptor %q", d.Digest)
-			}
-			if into.Images == nil {
-				into.Images = make(map[string]bundle.Image)
-			}
-			into.Images[componentName] = bundle.Image{
-				BaseImage: bundle.BaseImage{
-					Image:     refFamiliar,
-					ImageType: imageType,
-					MediaType: d.MediaType,
-					Size:      uint64(d.Size),
-				},
-			}
-		default:
-			return fmt.Errorf("invalid CNAB descriptor type %q in descriptor %q", descriptorType, d.Digest)
-		}
+func makeDescriptor(baseImage bundle.BaseImage, targetReference reference.Named, relocationMap relocation.ImageRelocationMap) (ocischemav1.Descriptor, error) {
+	relocatedImage, ok := relocationMap[baseImage.Image]
+	if !ok {
+		return ocischemav1.Descriptor{}, fmt.Errorf("unknown invocation image %q in the relocation map", baseImage.Image)
```

Comment:
The wording is a little off:
```suggestion
		return ocischemav1.Descriptor{}, fmt.Errorf("invocation image %q not present in the relocation map", baseImage.Image)
```